src/git.jl

```diff
@@ -1,32 +1,56 @@
 function _git_cmd(str::AbstractString;
                   adjust_PATH::Bool = true,
                   adjust_LIBPATH::Bool = true)
-    git_path, env_mapping = _env_mapping(; adjust_PATH, adjust_LIBPATH)
-    return addenv(`$(git_path) $(split(str))`, env_mapping)
+    cmd = git(; adjust_PATH, adjust_LIBPATH)
+    return `$(cmd) $(split(str))`
 end
 
 macro git_cmd(ex)
     return _git_cmd(ex)
 end
 
-function _gitrepl_parser(repl_input::AbstractString)
-    return quote
-        GitCommand.git() do git
-            repl_input = $(Expr(:quote, repl_input))
-            run(`$(git) $(split(repl_input))`)
-            return nothing
-        end
+function git(; adjust_PATH::Bool = true, adjust_LIBPATH::Bool = true)
+    if Sys.iswindows()
+        return Git_jll.git(; adjust_PATH, adjust_LIBPATH)
     end
-end
 
-function git(;
-             adjust_PATH::Bool = true,
-             adjust_LIBPATH::Bool = true)
-    git_path, env_mapping = _env_mapping(; adjust_PATH, adjust_LIBPATH)
-    return addenv(`$(git_path)`, env_mapping...)
+    root = Git_jll.artifact_dir
+
+    libexec = joinpath(root, "libexec")
+    libexec_git_core = joinpath(libexec, "git-core")
+
+    share = joinpath(root, "share")
+    share_git_core = joinpath(share, "git-core")
+    share_git_core_templates = joinpath(share_git_core, "templates")
+
+    libpaths = [
+        dirname(Expat_jll.libexpat_path),
+        dirname(Gettext_jll.libgettext_path),
+        dirname(LibCURL_jll.libcurl_path),
+        dirname(Libiconv_jll.libiconv_path),
+        dirname(XML2_jll.libxml2_path),
+        get(ENV, JLLWrappers.LIBPATH_env, "")
+    ]
+    filter!(x -> !isempty(x), libpaths)
+    newlibpath = join(libpaths, _separator())
+
+    ssl_cert = joinpath(dirname(Sys.BINDIR), "share", "julia", "cert.pem")
+
+    env_mapping = Dict{String,String}()
+    env_mapping["GIT_EXEC_PATH"] = libexec_git_core
+    env_mapping["GIT_SSL_CAINFO"] = ssl_cert
+    env_mapping["GIT_TEMPLATE_DIR"] = share_git_core_templates
+    env_mapping[JLLWrappers.LIBPATH_env] = newlibpath
```

Comment:
Actually, you don't even need to set this, it's already in `Git_jll.git()`.  I think at this point you don't need to load `JLLWrappers` either.

Reply:
I removed `env_mapping["GIT_SSL_CAINFO"] = ssl_cert` and `env_mapping[JLLWrappers.LIBPATH_env] = newlibpath`, and all the tests still pass.

Unfortunately, if I remove the `env_mapping["GIT_EXEC_PATH"] = libexec_git_core` line, the tests fail.

If I remove the `env_mapping["GIT_TEMPLATE_DIR"] = share_git_core_templates` line, I get the warning:
```
warning: templates not found in /workspace/destdir/share/git-core/templates
```

Reply:
Nope, I need `env_mapping["GIT_SSL_CAINFO"] = ssl_cert` - otherwise, on CI, I get:
```
Cloning into 'GitCommand.jl'...
fatal: unable to access 'https://github.com/JuliaVersionControl/GitCommand.jl/': Cert verify failed: BADCERT_NOT_TRUSTED
```

Reply:
Perhaps GitHub interface isn't clear, but this comment refers only to line 43 :slightly_smiling_face: 
```julia
    env_mapping[JLLWrappers.LIBPATH_env] = newlibpath 
```

Reply:
Ohhhhh. Yes I successfully removed that line.

GitHub showed four lines, so I got confused 😂